src/ucs/sys/sys.c

```diff
@@ -1648,3 +1649,21 @@ ucs_status_t ucs_sys_read_sysfs_file(const char *dev_name,
 
     return UCS_OK;
 }
+
+const char *ucs_sys_dmi_product_name()
+{
+    static char product_name[128] = {0};
+    static int initialized        = 0;
+    ssize_t nread;
+
+    if (!initialized) {
+        initialized = 1;
+        nread       = ucs_read_file_str(product_name, sizeof(product_name), 1,
+                                        UCS_DMI_PRODUCT_NAME_FILE);
+        if (nread >= 0) {
+            ucs_strtrim(product_name);
+        }
```

Comment:
else do we dump something like `N/A` or empty string?

Reply:
Dump `N/A` in case of failure.